cliquet/storage/generators.py

```diff
@@ -11,7 +11,7 @@ class Generator(object):
     resource level to validate record id in requests paths.
     """
 
-    regexp = r'^[a-zA-Z0-9\-]+$'
+    regexp = r'^[a-zA-Z0-9][a-zA-Z0-9_-]*$'
```

Comment:
Are you sure there is no need for blackslash ?

...would definitely require a small test :) Especially to assert the difference between first and other chracacters